pkg/ccl/changefeedccl/metrics.go

```diff
@@ -149,6 +152,27 @@ func (m *sliMetrics) getBackfillCallback() func() func() {
 	}
 }
 
+// getBackfillRangeCallback returns a backfillRangeCallback that is to be called
+// at the beginning of a backfill with the number of ranges that will be scanned
+// and returns a two callbacks to decrement the value until all ranges have
+// been emitted or clear the number completely if the backfill is cancelled.
+// Note: dec() should only be called as many times as the initial value, and
+// clear() should be called when there will never be another dec() call.
+func (m *sliMetrics) getBackfillRangeCallback() func(int64) (func(), func()) {
+	return func(initial int64) (dec func(), clear func()) {
+		remaining := initial
+		dec = func() {
+			atomic.AddInt64(&remaining, -1)
+			m.BackfillPendingRanges.Dec(1)
+		}
+		clear = func() {
+			atomic.AddInt64(&remaining, -remaining)
+			m.BackfillPendingRanges.Dec(remaining)
```

Comment:
That doesn't look right.  I don't think you want to update remaining prior to Dec, do you?